packages/react-reconciler/src/__tests__/ReactInterleavedUpdates-test.js

```diff
@@ -0,0 +1,129 @@
+let React;
+let ReactNoop;
+let Scheduler;
+let startTransition;
+let useState;
+let useEffect;
+
+describe('ReactInterleavedUpdates', () => {
+  beforeEach(() => {
+    jest.resetModules();
+
+    React = require('react');
+    ReactNoop = require('react-noop-renderer');
+    Scheduler = require('scheduler');
+    startTransition = React.unstable_startTransition;
+    useState = React.useState;
+    useEffect = React.useEffect;
+  });
+
+  function Text({text}) {
+    Scheduler.unstable_yieldValue(text);
+    return text;
+  }
+
+  test('update during an interleaved event is not processed during the current render', async () => {
```

Comment:
We already have a smattering of tests that cover these cases but they were kinda old and mostly pre-hooks, so I wrote these extra ones. Duplicative but still helpful when refactoring.